### ISSUE TITLE
Improve when `protected_regions` are created

### ISSUE THREAD

**Issue by kaste** [SublimeLinter/SublimeLinter#1697]
The sole purpose was cooperation with `GitGutter` iirc which worked 😐 at best.

Since then, we have git support in Sublime which actually draws the gutter stuff for us. I actually still use `GitGutter` (for the tooltips) which cooperates with the Sublime thing 😗.

tl;dr I don't think we need this anymore.

**Comment by braver** [comment]
Good point. Does brackethighlighter respect our protected_regions? Probably not. So yeah, this should be removed.

**Comment by kaste** [comment]
With the `protected_regions` GitGutter controls if it shows the `git diff` tooltip.  T.i. if I remove this marker I don't get our line report on hover. 

Actually, we should provide `protected_regions` only if `show_hover_line_report` is `True`.  Since nobody reported that as a bug ever, it may be that the setting `show_hover_line_report` is not needed.

**Comment by braver** [comment]
I would not be surprised if almost everyone keeps the default setting here, ie. enable `show_hover_line_report`. And maybe the few times our gutter icon conflicts with GitGutter they shrug. 

The important thing was the icon though. We felt that our icon was more important than theirs. But I think their popup is maybe more important than ours, because we have about 5 alternatives to the popup, and GitGutter doesn't.

**Comment by braver** [comment]
This also assumes that our tooltip is more important than GitGutter’s, which I’m not sure is true (for everyone always). I think we should either leave the functionality as is (it works when it works and apparently people don’t have big problems with it), or get rid of it. I don’t think it works well enough to warrant trying to make it smarter or enhancing it in any way, and people are not demanding we do anything extra here. 

I personally like the idea of taking out the protected regions because it doesn’t work that well and there are other ways of getting that info if another package steals “our” region. And I like taking features away 😈

**Comment by kaste** [comment]
I sure fix the bug here, since we now know it, and don't set protected regions if `show_hover_line_report` is `False`.  

Other than that it is an UI thing. When we draw things which look like "hoverable" then we also show the tooltip, maybe even if it is probably not that useful.  I can also imagine that we first did the gutter tooltips and then the setting was added because user actually found the line report not useful and wanted rather nothing or the GitGutter thing.

From my point of view, hard to see any benefit from the line tooltip. I actually use the error tooltip which comes naturally; you see an underline, and have your hand already on the mouse so you just approach it, and hover.  Don't know why you should do this on the line; and if you have multiple errors on the line, hard to tell for sure which is which. 

On the other hand, if I hover over such an icon (from us) and then GitGutter appears...  I already made a quick fix here, and it sure feels/looks strange. 

Probably or maybe the best experience is to show the tooltip when the icon is visible without any setting.  You either have icons, or just don't draw them.  But I have to try as well, if I actually like it to have no icons. Don't think I ever tried that.  

Or I just hide the icons on hover which would make it very clear that the tooltip does not come "from" the SublimeLinter icon.

**Comment by braver** [comment]
Sounds good, seems like some experimentation is needed here.

**Comment by braver** [comment]
Takeaways:

- we should not make protected regions if the user disabled the line tooltips
- we should keep the protected regions because it’s weird to hover over our icon and get someone else’s tooltip, and if the user prefers the other tooltips at all times we have the setting
- unless we drop the line tooltips altogether, which would be fine by me, but we need a better reason to do so

**Comment by kaste** [comment]
There is some confusion here.  Our protected regions denote where *we* draw icons.  GitGutter has two features around that: (a) it does not overwrite our icons -- giving SL's icons a higher priority -- and (b) it does not show tooltips on these regions -- giving SL's tooltip a higher priority.  Both features can be controlled in GitGutter via separate settings.  So, whether we show tooltips here in SL land is not important.  One can just turn off our tooltips, and then configure GitGutter to draw theirs, and have the desired outcome.  

Closing.  Nothing to do.
